### PR TITLE
Code cleanup: remove unused db attrs, dead MoM code

### PR DESCRIPTION
**Pull request by ashwathraop**
#### Describe Bug or Feature

Removal of unused attributes from DB and dead code from MoM.

#### Describe Your Change

- Removed most of the unused DB attributes.
- Removed dead code from MoM for the platforms like CSA, BGL and Darwin

### CODE REVIEW

**Comment by ashwathraop** [comment]
No new surprises. Only known issues:
image

File: .gitignore [hunk 1]
```diff
@@ -271,3 +271,4 @@ src/lib/Libpbs/pbs.pc
 *.cpio
 *.rpm
 *.deb
+.clang-format
```
Comment (hirenvadalia) [inline .gitignore:274]: I would say don't do this as there is already plan to add this file in repo

File: buildutils/pbs_mach_type [hunk 2]
```diff
@@ -66,8 +66,6 @@ test "$1" = "-v" && verbose=yes
 
 case "$OS" in
 
-  darwin*) mach=darwin ;;
```
Comment (hirenvadalia) [inline buildutils/pbs_mach_type:69]: I think we should remove this file as it has only one platform now "linux"

File: src/resmom/linux/mom_mach.c [hunk 3]
```diff
@@ -3883,213 +484,6 @@ dep_cleanup(void)
...
-
-/**
- * @brief
- *	Don't need any periodic procsessing except in some special cases.
- *
- * @return	Void
- *
- */
-void
-end_proc(void)
```
Comment (hirenvadalia) [inline src/resmom/linux/mom_mach.c:3920]: put its declaration in mom_func.h under #ifdef WIN32

File: src/server/req_manager.c [hunk 4]
```diff
@@ -3546,34 +3541,15 @@ struct batch_request *preq;
 			}
 		}
 
-		free(problem_nodes);		/*maybe problem malloc failed */
+		free(problem_nodes);		/* maybe problem malloc failed */
 		problem_nodes = NULL;
 
-		if (problem_cnt) {		/*reply has already been sent  */
+		if (problem_cnt) {		/* reply has already been sent */
 			return ;
```
Comment (hirenvadalia) [inline src/server/req_manager.c:3548]: remove space after return

File: src/server/req_manager.c [hunk 5]
```diff
@@ -3546,34 +3541,15 @@ struct batch_request *preq;
...
-		if ((presc != NULL) &&
-			(presc->rs_value.at_flags & ATR_VFLAG_SET)) {
-			if (strcmp(presc->rs_value.at_val.at_str, BLUEGENE) == 0)
-				rc = 1;
-		}
-	}
-	have_blue_gene_nodes = rc;
-
-	reply_ack(preq);		/*request completely successful*/
+	reply_ack(preq);		/* request completely successful */
```
Comment (hirenvadalia) [inline src/server/req_manager.c:3552]: I don't think we need comment on this line

File: src/tools/pbs_probe.c [hunk 6]
```diff
@@ -446,7 +446,7 @@ static char exinc[][80] = {
 
 static char exlib[][80] = {
 	/* 00 */ "lib/libattr.a",
-	/* 01 */ "lib/libcmds.a",	/* this library no longer exists */
+	/* 01 */ "SLOT_AVAILABLE",
```
Comment (hirenvadalia) [inline src/tools/pbs_probe.c:449]: did u tested pbs_probe after this change? I think put dummy string will fail

File: src/tools/pbs_upgrade_job.c [hunk 7]
```diff
@@ -103,7 +103,6 @@ typedef struct jobfix_PRE19
 	int ji_state;		/* internal copy of state */
 	int ji_substate;	/* job sub-state */
 	int ji_svrflags;	/* server flags */
-	int ji_numattr;		/* not used */
```
Comment (hirenvadalia) [inline src/tools/pbs_upgrade_job.c:106]: Don't we need to change this tool to handle changes we did in job fix area as part of this PR?

File: src/resmom/linux/mom_mach.c (hunk 3 above)
Comment (hirenvadalia) [reply src/resmom/linux/mom_mach.c:3920]: Also remove `set_mach_vars` declaration from mom_func.h

**Pull request closed without merging**